tests/common/mock_cgroup_commands.py

```diff
@@ -151,14 +171,33 @@ def mock_open(path, *args, **kwargs):
     def mock_path_exists(path):
         return original_path_exists(get_mapped_path(path))
 
+    data_files = [
+        (os.path.join(data_dir, 'init', 'walinuxagent.service'), UnitFilePaths.walinuxagent),
+        (os.path.join(data_dir, 'init', 'azure.slice'), UnitFilePaths.azure),
+        (os.path.join(data_dir, 'init', 'azure-vmextensions.slice'), UnitFilePaths.vmextensions)
+    ]
+
+    def add_data_file(source, target):
+        shutil.copyfile(source, get_mapped_path(target))
+
     builtin_popen = "__builtin__.open" if sys.version_info[0] == 2 else "builtins.open"
-    with patch("azurelinuxagent.common.cgroupapi.subprocess.Popen", side_effect=mock_popen) as patcher:
```

Comment:
For some reason i can't remember i was returning the Popen patcher but there are no more uses of it in the tests. Now the return value is the function itself, which is used to attach the properties on lines 191-198 in the new code